src/nvim/edit.c

```diff
@@ -2980,16 +2987,17 @@ static int ins_compl_prep(int c)
     compl_get_longest = (strstr((char *)p_cot, "longest") != NULL);
     compl_used_match = TRUE;
 
-    if (strstr((char *)p_cot, "noselect") != NULL) {
-      compl_no_insert = FALSE;
-      compl_no_select = TRUE;
-    } else if (strstr((char *)p_cot, "noinsert") != NULL) {
-      compl_no_insert = TRUE;
-      compl_no_select = FALSE;
-    } else {
-      compl_no_insert = FALSE;
-      compl_no_select = FALSE;
-    }
+  }
+
+  if (strstr((char *)p_cot, "noselect") != NULL) {
+    compl_no_insert = FALSE;
+    compl_no_select = TRUE;
+  } else if (strstr((char *)p_cot, "noinsert") != NULL) {
+    compl_no_insert = TRUE;
+    compl_no_select = FALSE;
```

Comment:
It is not exclusive option.
User can set both values.
But `compl_no_insert` and `compl_no_select` are exclusive variables.

So "else if" is needed.